packages/web-app/src/components/common/Maps/common/LayersControl.jsx

```diff
@@ -20,7 +21,21 @@ const LayersControl = ({
         key={layer.name}
         checked={layer.name === initialLayerChecked}
         name={layer.name}>
-        <TileLayer attribution={layer.attribution} url={layer.url} />
+        {layer.type === 'WMTS' && (
+          <TileLayer
+            attribution={layer.attribution}
+            url={layer.url}
+            maxZoom={layer.maxZoom ? layer.maxZoom : 22}
+            maxNativeZoom={layer.maxNativeZoom ? layer.maxNativeZoom : 22}
+          />
+        )}
+        {layer.type === 'WMS' && (
+          <WMSTileLayer
+            attribution={layer.attribution}
+            layers={layer.layers}
+            url={layer.url}
+          />
+        )}
       </LeafletLayersControl.BaseLayer>
```

Comment:
The layers list is also used in [PolygonMap.jsx](https://github.com/GrottoCenter/grottocenter-front/blob/develop/packages/web-app/src/components/appli/EntitiesForm/Massif/PolygonMap.jsx) and [MapMarkerSelector.jsx](https://github.com/GrottoCenter/grottocenter-front/blob/develop/packages/web-app/src/components/appli/EntitiesForm/utils/MapMarkerSelector.jsx)

Is it possible to make the change to take into account the layer type also there? 
Or may be it can be factored to avoid code duplication?

Reply:
I will try to refactor to use the same base layer component for all maps of grottocenter...